snuba_sdk/function.py

```diff
@@ -17,7 +17,7 @@ class InvalidFunction(InvalidExpression):
     pass
 
 
-alias_re = re.compile(r"^[a-zA-Z](\w|\.)+$")
+alias_re = re.compile(r"^[a-zA-Z]([^\s`'\"])*$")
```

Comment:
Which characters was the product using that we were not allowing?
This seem very permissive now as you can declare an alias `f(x)` when `f(x)` is a valid expression otherwise you probably do not want to shadow with an alias

Reply:
I updated the PR comment. This is inline with what we allowed as aliases in the legacy syntax, since we escape aliases with backticks.

Reply:
Right but in the old syntax the structure of the query itself (it was json) defined the tokens so, except when the users abused the aggregation field, you could not have a conflict between `f(x)` function call and `f(x)` alias as the function call had a json structure.

I don't think we should stick to that behavior for the sake of allowing a behavior that was ok in the old language but would become a smell in this language.

Can we fix the alias names in the product when they use square brackets or parentheses ?

Reply:
Are these queries generated by the automatic translation from legacy to SnQL ?

Reply:
I don't know why  the automatic translation vs. use of SnQL matters, but they are using the SDK directly.

Clickhouse has no limitations on what can be an alias, as long as you escape the alias with backticks. Snuba similarly has no restrictions since we escape backticks and allow any other characters.

https://github.com/getsentry/snuba/blob/master/snuba/clickhouse/escaping.py#L7

I'm not sure what you suggest I do, if someone wants to have an alias of `f(x)` then that should be allowed by us.

Reply:
> I don't know why the automatic translation vs. use of SnQL matters,
It matters because if they are using snql directly they can just change their alias right away, if it was an automatic translation, by preventing this, we would have made a valid query (in the old system) that had no reason to be rejected invalid.

> if someone wants to have an alias of f(x) then that should be allowed by us.
That is what I disagree on. I think it is a bad practice we can prevent.

Reply:
Also I think it may not be trivial to tell alias references (in the case of `f(x)` or `tags[asd]` used as aliases) apart from a function call or a tag reference during parsing unless we require them to be escaped.